allauth/socialaccount/providers/auth0/provider.py

```diff
@@ -9,7 +9,10 @@ def get_avatar_url(self):
 
     def to_str(self):
         dflt = super(Auth0Account, self).to_str()
-        return self.account.extra_data.get("name", dflt)
+        email = self.account.extra_data.get("email")
+        username = self.account.extra_data.get("username")
+        name = self.account.extra_data.get("name")
+        return email or username or name or dflt
```

Comment:
Thought: all these `to_str()` are almost the same. Given that the world is moving towards a standardized OIDC payload, we could also fold one implementation into a new to be created `OAuth2ProviderAccount`, and inside unify the logic?

Reply:
I changed `to_str` in the base class `ProviderAccount` to look at `email`, `username` and `name`. If you think it is better to create a new class `OAuth2ProviderAccount`, we can do that.

Reply:
Ok, this is fine, but I do think that many of the `to_str()` should be revisitted, as most of them likely became superfluous.

Reply:
We could delete most of the `to_str` implementations, but I don't think we should leave them as is, as many of them return the full name of the user, rather than a unique identifier (like a username or email address).

Reply:
That is what I meant -- delete them, unless they are really doing something specific (as is the case with Apple).